src/resources/parser/glb-parser.js

```diff
@@ -1571,11 +1571,11 @@ var applySampler = function (texture, gltfSampler) {
     };
 
     if (texture) {
-        gltfSampler = gltfSampler || defaultSampler;
-        texture.minFilter = getFilter(gltfSampler.minFilter);
-        texture.magFilter = getFilter(gltfSampler.magFilter);
-        texture.addressU = getWrap(gltfSampler.wrapS);
-        texture.addressV = getWrap(gltfSampler.wrapT);
+        Object.assign(sampler, gltfSampler || { });
+        texture.minFilter = getFilter(sampler.minFilter);
```

Comment:
It'd be probably nicer to change getFilter and getWrap to take a default value to return when undefined is passed in. Something like:
`texture.minFilter = getFilter(sampler.minFilter, FILTER_LINEAR);`
This would be more readable than the `minFilter: 9987,`
Or let it return undefined and supply engine default here
`texture.minFilter = getFilter(sampler.minFilter) || FILTER_LINEAR;`

Reply:
Agree: https://github.com/playcanvas/engine/pull/3015